pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/data/BibDataAsMarcRecord.scala

```diff
@@ -41,6 +41,11 @@ object SierraMarcDataConversions extends SierraMarcDataConversions {}
 class BibDataAsMarcRecord(bibData: SierraBibData)
     extends MarcRecord
     with SierraQueryOps {
+
+  // BibData doesn't have a leader or control fields, so we provide empty values
```

Comment:
This is one of the tricksy differences between Sierra and Ebsco data.

Some of the Sierra JSON varFields are XML controlFields.  e.g. [008](https://c783b93d8b0b4b11900b5793cb2a1865.eu-west-1.aws.found.io:9243/s/sierra/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-15m,to:now))&_a=(columns:!(),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d5ea4b8c-f58d-409c-8fb7-8a9973ea67f7,key:varField.marcTag,negate:!f,params:(query:'008'),type:phrase),query:(match_phrase:(varField.marcTag:'008')))),index:d5ea4b8c-f58d-409c-8fb7-8a9973ea67f7,interval:auto,query:(language:lucene,query:''),sort:!())).

In the context of our pipeline, I don't think that the distinction between controlField vs dataField vs fixedField vs varField is particularly helpful.

leader does exist in the Sierra data, as a [varField with fieldTag="_"](https://c783b93d8b0b4b11900b5793cb2a1865.eu-west-1.aws.found.io:9243/s/sierra/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-15m,to:now))&_a=(columns:!(),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d5ea4b8c-f58d-409c-8fb7-8a9973ea67f7,key:varField.fieldTag,negate:!f,params:(query:_),type:phrase),query:(match_phrase:(varField.fieldTag:_)))),index:d5ea4b8c-f58d-409c-8fb7-8a9973ea67f7,interval:auto,query:(language:lucene,query:''),sort:!())). (🤷)

Reply:
There is a distinction in that control fields need to be parsed differently from other fields (they are either byte positional or taken as a whole) which is the motivation for representing them in MarcRecord and knock-on having to deal with the fields here.

Having meaningful values for leader and control fields may be helpful to the project of generating works from our MARC interpretation solely where-ever possible as it could allow us to more thoroughly ignore the Sierra specifics (outside of Wellcome Collection cataloguing practises).

The `CollectionPath` transformer goes further in this direction here: https://github.com/wellcomecollection/catalogue-pipeline/pull/2639/files#diff-16d300d7619667c043ac6fcff53271c736459a6ee6a02139d7a26640c79c8602R57

What do you think?